src/System.Management.Automation/namespaces/LocationGlobber.cs

```diff
@@ -1230,12 +1225,9 @@ internal string GetProviderPath(string path, CmdletProviderContext context)
             drive = null;
 
             // First check to see if the path starts with a ~ (home)
-            if (IsHomePath(path))
+            using (s_pathResolutionTracer.TraceScope("Resolving HOME relative path."))
             {
-                using (s_pathResolutionTracer.TraceScope("Resolving HOME relative path."))
-                {
-                    path = GetHomeRelativePath(path);
-                }
+                path = GetHomeRelativePath(path);
```

Comment:
Same here.

Reply:
I left this comment as a clarification of intentions.